src/freedv_api.c

```diff
@@ -431,6 +438,17 @@ void freedv_datatx (struct freedv *f, short mod_out[]) {
     }
 }
 
+void freedv_datacomptx(struct freedv *f, COMP mod_out[])
+{
+    assert(f != NULL);
+    if (FDV_MODE_ACTIVE( FREEDV_MODE_2400A, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_2400B, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_800XA, f->mode)) {
+        short out[f->n_nom_modem_samples];
+        freedv_tx_fsk_data(f, out);
+	for (int i = 0; i < f->n_nom_modem_samples; i++)
```

Comment:
@JeroenVreeken sorry about the delay in looking at this.  I saw the ctests had failed so didn't take a look until now.  Pls feel free to ping me if I'm slow in responding to an open PR.

```freedv_tx_fsk_data()``` returns real samples, so making a complex version is not as simple as just filling in the real part.   That would generate a two sided spectrum.  While you could write a ctest that passes, it's still a two sided signal and not complex/single sided.

I think this is possible to use the complex FSK modulator function for 800XA and 2400A but it's not meaningful for 2400B which operates on baseband real samples either side of an analog FM modem.

Reply:
Looking at it again I indeed think this pull request is wrong...
I actually got confused by freedv-gui a bit as it used comptx for everything except 800XA....
The exception is actually correct, but it should do it for more modes....